cishouseholds/derive.py

```diff
@@ -107,7 +107,11 @@ def assign_filename_column(df: DataFrame, column_name_to_assign: str) -> DataFra
     df
     column_name_to_assign
     """
-    return df.withColumn(column_name_to_assign, F.input_file_name())
+    df = df.withColumn("temp_path", F.input_file_name())
+
+    df = df.withColumn(column_name_to_assign, F.regexp_replace("temp_path", r"(?<=\/{2})(\w+|\d+)(?=\/{1})", ""))
```

Comment:
Update to only edit path where there is not already three slashes (`///`) at the beginning of the path (e.g. file:///)

Reply:
Updated to check for three slashes